R/sample.R

```diff
@@ -134,7 +134,7 @@ initial_parameters <- function(initial, model, rng, call = NULL) {
                           arg = "initial", call = environment())
     ## Really this would just be from the prior; we can't directly
     ## sample from the posterior!
-    initial <- lapply(rng, function(r) model$direct_sample(r))
+    initial <- lapply(rng, function(r) direct_sample_within_domain(model, r))
```

Comment:
Should max_attempts be configurable here, from a param to initial_parameters?

Reply:
I hope that it's never needed, tbh. I'm thinking we'll see if anyone ever hits it. But if a model has failed in 100 tries it's never going to succeed I expect and it's mostly there to stop an infinite loop while not bailing too early